src/CardPage.js

```diff
@@ -0,0 +1,268 @@
+                  <div className='demo-card__media-title mdc-typography--subtitle2'>
+                    Vacation Photos
+                  </div>
+                </div>
+              </div>
+            </div>
+            <div className='mdc-card__actions mdc-card__action-icons'>
+              <i className='mdc-icon-toggle material-icons mdc-card__action mdc-card__action--icon'
+                 tabIndex='0'
+                 role='button'
+                 aria-pressed='false'
+                 aria-label='Add to favorites'
+                 title='Add to favorites'
+                 data-toggle-on='{"content": "favorite", "label": "Remove from favorites"}'
+                 data-toggle-off='{"content": "favorite_border", "label": "Add to favorites"}'
+                 ref={this.initIconToggle}>
+                favorite_border
+              </i>
+              <i className='mdc-icon-toggle material-icons mdc-card__action mdc-card__action--icon'
+                 tabIndex='0'
+                 role='button'
+                 aria-pressed='false'
+                 aria-label='Add bookmark'
+                 title='Add bookmark'
+                 data-toggle-on='{"content": "bookmark", "label": "Remove bookmark"}'
+                 data-toggle-off='{"content": "bookmark_border", "label": "Add bookmark"}'
+                 ref={this.initIconToggle}>
+                bookmark_border
+              </i>
+              <i className='material-icons mdc-card__action mdc-card__action--icon mdc-ripple-surface'
+                 tabIndex='0'
+                 role='button'
+                 data-mdc-ripple-is-unbounded
+                 title='Share'
+                 ref={this.initRipple}>
+                share
+              </i>
+            </div>
+          </div>
+
+          <div className='mdc-card demo-card demo-card--music'>
+            <div className='mdc-card__primary-action demo-card__primary-action' tabIndex='0' ref={this.initRipple}>
+              <div className='demo-card__music-row'>
+                <div className='mdc-card__media mdc-card__media--square demo-card__media demo-card__media--music'
+                     style={{backgroundImage: `url("${cardMedia1x1Img}")`}}></div>
+                <div className='demo-card__music-info'>
+                  <div className='demo-card__music-title mdc-typography--headline5'>Rozes</div>
+                  <div className='demo-card__music-artist mdc-typography--body2'>Under the Grave</div>
+                  <div className='demo-card__music-year mdc-typography--body2'>(2016)</div>
+                </div>
+              </div>
+            </div>
+            <hr className='mdc-list-divider'/>
+            <div className='mdc-card__actions'>
+              <div className='mdc-card__action-buttons demo-card__action-buttons--text-only'>Rate this album</div>
+              <div className='mdc-card__action-icons'>
```

Comment:
These `<i>`s can also become a self contained component so there's not as much repetition.

```
class CardActionIcon extends Component {
  ripple = null;
  
  componentWillUnmount() {
    this.ripple.destroy();
  }
  render() {
    return (
      <i className='material-icons demo-card__action-icon--star mdc-ripple-surface'
         tabIndex='0'
         role='button'
         title={this.props.title}
         data-mdc-ripple-is-unbounded
         ref={(surfaceEl) => this.ripple = new MDCRipple(surfaceEl)}>
         star_border
      </i>
    );
  }
}

```


Then the markup would become:

```
<div className='mdc-card__action-icons'>
  <CardActionIcon title='1 star'/>
  <CardActionIcon title='2 star'/>
  <CardActionIcon title='3 star'/>
  <CardActionIcon title='4 star'/>
  <CardActionIcon title='5 star'/>
</div>